webrtc/src/rtp_transceiver/rtp_sender/mod.rs

```diff
@@ -454,24 +539,69 @@ impl RTCRtpSender {
 
         self.replace_track(None).await?;
 
-        {
-            let stream_info = self.stream_info.lock().await;
+        let encodings = self.track_encodings.read().await;
+        for te in encodings.iter() {
+            let stream_info = te.stream_info.lock().await;
             self.interceptor.unbind_local_stream(&stream_info).await;
+            te.srtp_stream.close().await?
         }
-
-        self.srtp_stream.close().await
+        Ok(())
     }
 
     /// read reads incoming RTCP for this RTPReceiver
     pub async fn read(&self, b: &mut [u8]) -> Result<(usize, Attributes)> {
-        self.internal.read(b).await
+        if let Some(encoding) = self.first_encoding().await? {
+            self.internal.read(&encoding, b).await
+        } else {
+            Err(Error::ErrInterceptorNotBind)
+        }
+    }
+
+    async fn encoding_for_rid(&self, rid: &str) -> Option<Arc<TrackEncoding>> {
+        let encodings = self.track_encodings.read().await;
+        for e in encodings.iter() {
+            if let Some(track) = &*e.track.lock().await {
+                if track.rid().map_or(false, |r| r == rid) {
+                    return Some(e.clone());
+                }
+            };
+        }
+        None
+    }
+
+    async fn first_encoding(&self) -> Result<Option<Arc<TrackEncoding>>> {
+        let encodings = self.track_encodings.read().await;
+        return Ok(encodings.first().map(|x| (*x).clone()));
+    }
+
+    pub async fn read_simulcast(&self, b: &mut [u8], rid: &str) -> Result<(usize, Attributes)> {
+        if let Some(encoding) = self.encoding_for_rid(rid).await {
+            self.internal.read(&encoding, b).await
+        } else {
+            Err(Error::ErrRTPSenderNoTrackForRID)
+        }
+    }
+
+    pub async fn read_simulcast_rtcp(
```

Comment:
This method is weird, RTCP isn't tied to simulcast, I think we should remove it. Am I missing something?

Reply:
This is a pion-ism, I'm not using it in my end-user code (although obviously I use ```read_simulcast```. I don't care if it stays or goes, although given that Pion added it it's either because they're missing something or we're missing something.

Reply:
This comes down to that duplicated RTCP infrastructure really, looking at it. This is Pion all the way down, we can go another way if we want but it seems like needless digression unless there are big problems with this approach.

Reply:
We don't have to do everything Pion does though, we've already diverged in several areas. Fundamentally is there a reason would use `read_simulcast_rtcp` of just `read_rtcp`? I can't think of any so I'd suggest this method goes

Reply:
Fair enough, I can't think of a reason it should exist either